cantools/subparsers/dump/formatting.py

```diff
@@ -0,0 +1,372 @@
+import textwrap
+
+from ...database.utils import start_bit
+
+
+class Colors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+def signal_tree_string(message, console_width=80, with_comments=False):
+    """Returns the message signal tree as a string.
+
+    """
+
+    def get_prefix(index, length):
+        if index < length - 1:
+            return '|   '
+        else:
+            return '    '
+
+    def add_prefix(prefix, lines):
+        return [prefix + line for line in lines]
+
+    def format_signal_line(signal_name):
+        siginst = message.get_signal_by_name(signal_name)
+        signal_name_line = signal_name
+
+        if with_comments:
+            com = []
+            if siginst.comment:
+                com.append(siginst.comment)
```

Comment:
I just noticed this: the ARXML file format supports multi-lingual comments, so the canonical way to access the comments is the `.comments` attribute of messages/signals. (which is a language string -> comment string dictionary. `.comment` is just the first entry of this dictionary.) If you don't consider it worthwhile to support such an exotic feature, I think this PR should go in as-is.

Reply:
Sounds like an additional PR to support that. I will keep it in the back of my head though :) Thanks for your review effort!